nomad/acl.go

```diff
@@ -2,15 +2,117 @@ package nomad
 
 import (
 	"fmt"
+	"net"
 	"time"
 
 	metrics "github.com/armon/go-metrics"
 	lru "github.com/hashicorp/golang-lru"
 	"github.com/hashicorp/nomad/acl"
+	"github.com/hashicorp/nomad/helper"
 	"github.com/hashicorp/nomad/nomad/state"
 	"github.com/hashicorp/nomad/nomad/structs"
 )
 
+// Authenticate extracts an AuthenticatedIdentity from the request context or
+// provided token. The caller can extract an acl.ACL, WorkloadIdentity, or other
+// identifying token to use for authorization.
```

Comment:
Note for reviewers: something that's a little quirky here is that because we're making a stale query against the state store, it's possible a follower gets a different value than the leader if an ACL token or alloc has just been created. Because we don't expect the follower to actually serve the request unless the user has specifically asked for a stale query, I think this ends up being ok but will produce slightly off metrics. (The race is also just really unlikely unless a follower is far behind.)

Reply:
Want to add this clarification to the doc string?

Reply:
Will do